### PR TITLE
[CI, OPAM] Compatible with MathComp 1.13.0 and dev

### PR DESCRIPTION
**Pull request by pi8027**

### CODE REVIEW

File: coq-mathcomp-multinomials.opam [hunk 1]
```diff
@@ -14,7 +14,7 @@ build: [
 depends: [
   "coq"                    {(>= "8.10" & < "8.15~") | = "dev"}
   "dune"                   {>= "2.5"}
-  "coq-mathcomp-ssreflect" {(>= "1.12" & < "1.13~") | = "dev"}
+  "coq-mathcomp-ssreflect" {(>= "1.12" & < "1.14~") | = "dev"}
   "coq-mathcomp-algebra"
   "coq-mathcomp-bigenough" {(>= "1.0" & < "1.1~") | = "dev"}
   "coq-mathcomp-finmap"    {(>= "1.5" & < "1.6~") | = "dev"}
```
Comment (pi8027) [inline coq-mathcomp-multinomials.opam:20]: I think we can simply remove the upper bound here (and `coq` above). Right? I mean:
```suggestion
  "coq-mathcomp-ssreflect" {>= "1.12"}
  "coq-mathcomp-algebra"
  "coq-mathcomp-bigenough" {>= "1.0"}
  "coq-mathcomp-finmap"    {>= "1.5"}
```

**Review by erikmd — APPROVED** [review]
Hi @pi8027!

> I think we can simply remove the upper bound(s) here

Indeed, even if `opam-coq-archive` maintainers prefer to have such upper bounds in the `released` opam file, I believe that's very OK for the multinomials.dev opam file.
(BTW you may also want to update this one:
https://github.com/coq/opam-coq-archive/blob/master/extra-dev/packages/coq-mathcomp-multinomials/coq-mathcomp-multinomials.dev/opam)

**Comment by palmskog** [comment]
The main reason we use the `dev` scheme in the in-repo opam files in a bunch of projects in Coq-community and elsewhere:
```
"coq-mathcomp-ssreflect" {(>= "1.12" & < "1.14~") | = "dev"}
```
... is to keep track of what stable versions of packages are currently tested in CI. Then, when a new version is released, one simply removes the `| = "dev"` part for the `released` opam package.

**Comment by pi8027** [comment]
@palmskog Thanks. I will revert.

**Comment by palmskog** [comment]
@pi8027 just to be clear, no requirement from me to revert here - just wanted to highlight the tradeoffs we did when we came up with the format. I also think it makes sense to skip upper bounds in `extra-dev` (to not have to update the package there all the time). But then one gets diverging packages in the repo and in `extra-dev`, which is not great either. We seem to lack automation to handle these things properly...

**Comment by strub** [comment]
@pi8027 The PR is read to be merged?

**Comment by pi8027** [comment]
@strub If you meant "ready", yes.

**Pull request merged**